docs/operations/versioning_policy.md

```diff
@@ -0,0 +1,279 @@
+# Gardener Versioning Policy
+
+## Goal
+
+- As a Garden operator I would like to define a clear Kubernetes version policy, which informs my users about deprecated or expired Kubernetes versions.
+- As an user of Gardener, I would like to get information which Kubernetes version is supported or in preview. I want to be able to get this information via API (cloudprofile) and also in the Dashboard.
+
+## Motivation
+
+The Kubernetes community releases **minor** versions roughly every three months and usually maintains **three minor** versions (the actual and the last two) with bug fixes and security updates. 
+Patch releases are done more frequently.
+Operators of Gardener are able to define their own Kubernetes version policy. 
+This gardener Versioning Policy makes it possible for operators to classify Kubernetes versions, while they are going through their "maintenance life-cycle".
+
+## Kubernetes Version Classifications
+
+An operator is able to classify Kubernetes versions differently while they go through their "maintenance life-cycle", starting with **preview**, **supported**, **deprecated**, and finally **expired**. 
+This information is programmatically available in the `CloudProfiles` of the Garden cluster as well as taken into consideration in the Dashboard. 
+Please also note, that Gardener keeps the control plane and the workers on the same Kubernetes version.
+
+For further explanation of the possible classifications, we assume that an operator wants to support four minor versions e.g. v1.16, v1.15, v1.14 and v1.13.
+
+- **preview:** After a fresh release of a new Kubernetes **minor** version (e.g. v1.17.0) the operator could tag it as _preview_ until he has gained sufficient experience. It will not become the default in the Gardener Dashboard until he promotes that minor version to _supported_, which could happen a few weeks later with the first patch version.
+
+- **supported:** The operator would tag the latest Kubernetes patch versions of the actual (if not still in _preview_) and the last three minor Kubernetes versions as _supported_ (e.g. v1.16.1, v1.15.4, v1.14.9 and v1.13.12). The latest of these becomes the default in the Gardener Dashboard (e.g. v1.16.1).
```

Comment:
Actually GEP-05 and this doc, both of them do not describe what is the semantic behind these classifications.
For example we have classification `preview`,  but we don't have classification `stable` (because if there is a `preview` which is considered the alpha/beta, I would expect to have the stable, such as the GKE release channels or CoreOS versions), . Does `supported` means `stable`? If yes, a `supported` thing can be a `stable` thing or a `deprecated` thing. The fact that it is `supported` does not makes it `stable`. Or `supported` is just reference that these versions are currently supported/patched by the Kubernetes community? Actually these questions are to the GEP-05 authors (I guess @vlerenc, @RaphaelVogel).

We could add explanations such as (for example):
- `preview` can contain unresolved issues and is not recommended for production usage
- `deprecated` can contain issues which are probably resolved in a supported version
- etc.

WDYT?

Reply:
I rephrased the docs & used your feedback - thanks!.  Feel free to check it again.

Reply:
Looks better now, thank you!